analyzers/tests/SonarAnalyzer.UnitTest/AnalysisContext/SonarAnalysisContextBaseTest.ShouldAnalyzeFile.cs

```diff
@@ -0,0 +1,99 @@
+﻿/*
```

Comment:
These test cases could have gone to the `ShouldAnalyzeTree` test cases. Basically we are also checking if the SyntaxTree should be analyzed, the difference is that we decide based on the file it contains the tree.